lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -115,18 +116,37 @@ public void writeBlock(ColumnChunk columnChunk)
         // write values
         primitiveValueWriter.write(columnChunk.getBlock());
 
-        // write definition levels
-        Iterator<Integer> defIterator = DefLevelIterables.getIterator(ImmutableList.<DefLevelIterable>builder()
-                .addAll(columnChunk.getDefLevelIterables())
-                .add(DefLevelIterables.of(columnChunk.getBlock(), maxDefinitionLevel))
-                .build());
-        while (defIterator.hasNext()) {
-            int next = defIterator.next();
-            definitionLevelWriter.writeInteger(next);
-            if (next != maxDefinitionLevel) {
-                currentPageNullCounts++;
+        if (columnChunk.getDefLevelIterables().isEmpty()) {
+            // write definition levels for flat data types
+            Block block = columnChunk.getBlock();
+            if (!block.mayHaveNull()) {
+                for (int position = 0; position < block.getPositionCount(); position++) {
+                    definitionLevelWriter.writeInteger(maxDefinitionLevel);
+                }
+            }
+            else {
+                for (int position = 0; position < block.getPositionCount(); position++) {
+                    byte isNull = (byte) (block.isNull(position) ? 1 : 0);
+                    definitionLevelWriter.writeInteger(maxDefinitionLevel - isNull);
```

Comment:
maxDefinitionLevel needs to be at least 1, right?

Reply:
No. For non-null flat values it is usually 0